src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -135,7 +149,8 @@ public static ValidationException validateKnnField(
         IndexMetadata indexMetadata,
         String field,
         int expectedDimension,
-        ModelDao modelDao
+        ModelDao modelDao,
+        VectorDataType expectedVectorDataType
```

Comment:
Since we are adding extra parameters to function , we will have make sure we are passing this extra param to all flow which is not required. Let's have overloaded function with extra params which makes sure that other flow is not affected and then refactor this method to avoid code duplicatation.

Reply:
updated the validateKnnField function to have vector data type check when it's non null, the flows will be verified by adding tests